ui/app/adapters/kv/data.js

```diff
@@ -85,39 +91,7 @@ export default class KvDataAdapter extends ApplicationAdapter {
         };
       })
       .catch((errorOrResponse) => {
-        const baseResponse = { id, backend, path, version };
```

Comment:
the subkeys response handles deleted records the same as `GET` requests do. I decided to pull this logic out into a helper so that it could be reused in both requests.